src/state/analysisGame/analysisGameReducer.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  AnalysisGame,
+  GenericAction,
+  TileValue,
+  Player,
+  Winner,
+} from '../AppState';
+import {
+  MOVE_BACKWARD_IN_HISTORY,
+  MOVE_FORWARD_IN_HISTORY,
+  SET_ANALYSIS_GAME,
+} from './analysisGameActions';
+import produce from 'immer';
+import { arePointsEqual, playerToTileValue } from '../../util';
+import { getWinResult } from '../../util/CheckBoard';
+import { getNewActiveBoards } from '../../util/ActiveBoards';
+
+const initialState: AnalysisGame = {
+  id: '',
+  board: [],
+  activeBoards: [],
+  game: {
+    isFinished: true,
+    winningPlayer: Winner.None,
+    currentPlayer: Player.Cross,
+  },
+  moves: [],
+  currentMove: 1,
+};
+
+const analysisGameReducer = (state = initialState, action: GenericAction) => {
+  switch (action.type) {
+    case SET_ANALYSIS_GAME: {
+      return action.payload;
+    }
+    case MOVE_FORWARD_IN_HISTORY: {
+      const newState = produce(state, (draftState) => {
+        const movesToMoveForward = action.payload;
+        draftState.currentMove += movesToMoveForward;
+
+        const moveIndex = state.moves.findIndex(
+          (m) => m.moveNumber === draftState.currentMove,
+        );
+        const lastPlayer = state.moves[moveIndex].player;
+        if (lastPlayer === Player.Circle) {
+          state.game.currentPlayer = Player.Cross;
+        } else {
+          state.game.currentPlayer = Player.Circle;
+        }
+
+        const movesToApply = state.moves.slice(
+          state.currentMove,
+          moveIndex + 1,
+        );
+        movesToApply.forEach((m) => {
+          const boardToChange = draftState.board.find((b) =>
+            arePointsEqual(b.position, m.boardPosition),
+          );
+          const tileToChange = boardToChange!.tiles.find((t) =>
+            arePointsEqual(t.position, m.tilePosition),
+          );
+          tileToChange!.value = playerToTileValue(m.player);
+
+          const winResult = getWinResult(boardToChange!.tiles);
+          if (winResult.isFinished) {
+            boardToChange!.value = playerToTileValue(
+              winResult.winningPlayer,
+              true,
+            );
+          }
+        });
+
+        const lastAppliedMove = movesToApply[movesToApply.length - 1];
+        const lastGameMove = draftState.moves[draftState.moves.length - 1];
+        const isLastAppliedMoveTheLastGameMove =
+          lastAppliedMove.moveNumber === lastGameMove.moveNumber;
+        if (isLastAppliedMoveTheLastGameMove) {
+          draftState.activeBoards = [];
+        } else {
+          draftState.activeBoards = getNewActiveBoards(
+            lastAppliedMove.tilePosition,
+            draftState.board,
+          );
+        }
+      });
```

Comment:
I will have a closer look at that later. Maybe this can be done easier or shorter

Reply:
The choosing of the new player could be done in it's one function. (and also be used in the gameReducer) But that won't make it much shorter/easier.

Let me know if you have any suggestions.

Reply:
I added a board game class on a different branch that will fix your concern:
https://github.com/ultimate-ttt/ultimate-ttt/blob/juim/test-board-class/src/state/analysisGame/analysisGameReducer.ts

(Also refactored the other logic to use the new class. Tests for the new class are not written yet and old ones obviously failing though)

Reply:
sounds good